### PR TITLE
docs: `mount dir as secret` cookbook

### LINKED ISSUE #4896: ✨Allow directories to be mounted as secrets
### What are you trying to do?

I'm trying to create and GPG-sign a built binary with Dagger. 

To do this, I need to mount my ~/.gnupg directory into my container and run `gpg --detach-sign --armor binary-name`.

The above command requires the following files (keyrings and trustdb) to create the signature:
- `~/.gnupg/pubring.kbx`
- `~/.gnupg/trustdb.gpg`
- `~/.gnupg/private-keys-v1.d/*`

Currently Dagger only permits mounting files as secrets. It's not possible to mount an entire directory hierarchy as a secret; each file has to be mounted manually.

Can Dagger allow directories to be mounted as secrets, as it does with files?

### Why is this important to you?

It will make it simpler to sign binary builds with GPG.

### How are you currently working around this?

_No response_

### PR DESCRIPTION
**Pull request by grouville**
Fixes #4896

⚠️ PUBLISH AFTER v0.8.0 RELEASE, depends on #5500 

Small cookbook example / wrapper with the `.with()` method showing how to mount directories securely

### CODE REVIEW

**Comment by grouville** [comment]
For reference purposes, it works perfectly: https://gist.github.com/grouville/8aae4c02c1036ebdfac196545f0cb3ac.

Here is the output: 
```shell
go run .
12: resolve image config for docker.io/library/alpine:3.17
12: > in from alpine:3.17
12: resolve image config for docker.io/library/alpine:3.17 DONE

34: pull docker.io/library/alpine:3.17
34: > in from alpine:3.17
34: resolve docker.io/library/alpine:3.17@sha256:e95676db9e4a4f16f6cc01a8915368f82b018cc07aba951c1bd1db586c081388
34: resolve docker.io/library/alpine:3.17@sha256:e95676db9e4a4f16f6cc01a8915368f82b018cc07aba951c1bd1db586c081388 [0.01s]
34: pull docker.io/library/alpine:3.17 DONE

31: exec chmod 700 /root/.gnupg CACHED
31: exec chmod 700 /root/.gnupg CACHED

32: exec apk add --no-cache gnupg CACHED
32: exec apk add --no-cache gnupg CACHED

30: exec chmod 700 /root/.gnupg/private-keys-v1.d CACHED
30: exec chmod 700 /root/.gnupg/private-keys-v1.d CACHED

29: exec gpg --import /root/.gnupg/public.key
29: [0.18s] gpg: WARNING: unsafe permissions on homedir '/root/.gnupg'
29: [0.18s] gpg: key 3CCB68B1170F9A36: "guigui <guigui@guigui.fr>" not changed
29: [0.18s] gpg: Total number processed: 1
29: [0.18s] gpg:              unchanged: 1
29: exec gpg --import /root/.gnupg/public.key DONE

28: exec gpg --import /root/.gnupg/private-keys-v1.d/private.key
28: [0.16s] gpg: WARNING: unsafe permissions on homedir '/root/.gnupg'
28: [0.16s] gpg: key 3CCB68B1170F9A36: "guigui <guigui@guigui.fr>" not changed
28: [0.16s] gpg: key 3CCB68B1170F9A36: secret key imported
28: [0.16s] gpg: Total number processed: 1
28: [0.16s] gpg:              unchanged: 1
28: [0.16s] gpg:       secret keys read: 1
28: [0.16s] gpg:   secret keys imported: 1
28: exec gpg --import /root/.gnupg/private-keys-v1.d/private.key DONE

27: exec gpg --detach-sign --armor /bin/ls
27: [0.16s] gpg: WARNING: unsafe permissions on homedir '/root/.gnupg'
27: [0.16s] gpg: Note: trustdb not writable
27: exec gpg --detach-sign --armor /bin/ls DONE

26: exec ls -l /bin/ls.asc
26: [0.15s] -rw-r--r--    1 root     root           228 Jul 24 14:32 /bin/ls.asc
26: exec ls -l /bin/ls.asc DONE
-rw-r--r--    1 root     root           228 Jul 24 14:32 /bin/ls.asc
```

The signature has been properly done ✅

File: docs/current/742989-cookbook.md [hunk 1]
```diff
@@ -625,6 +625,31 @@ Set the Hashicorp Vault URI, namespace, role and secret identifiers as host envi
 
 [Learn more](./guides/723462-use-secrets.md)
 
+### Mount directories as secrets in a container
+
+The following code listing demonstrates how to securely mount directories as secrets in a container. The directory structure will be exposed:
```
Comment (vikram-dagger) [inline docs/current/742989-cookbook.md:630]: `The directory structure will be exposed` - what does this mean? Do you mean that you can see the directory/file names, but not read the contents? Or something else?

**Review by vikram-dagger — APPROVED** [review]
@grouville nice work. I'm approving subject to an improvement regarding a sentence where the implication was not immediately clear to me - can you please add/clarify that point before merging? Thanks.

File: docs/current/742989-cookbook.md (hunk 1 above)
Comment (grouville) [reply docs/current/742989-cookbook.md]: Exactly, the content will be scrubbed, not the file names neither the directory structure. Updated accordingly 🙏

**Pull request merged**